thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -2662,6 +2662,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
     }
     void readAhead()
     {
+        LookAheadTimer t(slaveTimerStats, timeActivities);
```

Comment:
this means it will only update at the very end of reading the whole LHS input.
Perhaps should do every rows, it should call a t.commitTime() - that updates and reset startTime to current.

Reply:
But also, I'm not sure that what it's considering as lookahead time is correct here (not just the time it spend reading from the input). Is it right?
As it stands, it's going to consider the input read time + the rest of the logica here in readAhead (e.g. the matching, lookup etc.), so totalCycles + lookAheadCycles - inputCycles, could leave local cycles looking too big,

Reply:
I'm putting the lookahead around just the two inputStream->nextRow calls.  That should mean that the tracking is more frequent and it should exclude the time for the other logic.